### PR TITLE
[TTAHUB-1857] Update event permissions on BE and FE

### PR DESCRIPTION
**Pull request by AdamAdHocTeam**
## Description of change

Only a user with write permission and a assigned event role (poc, collab, or owner) can create or edit an event and session.

## How to test

Create a user who has write permission but is NOT assigned and a event role. They should only be able to view.

Create a user who has write permission and IS assigned a role. They should be able to create, read, update and delete that event and its sessions.

## Issue(s)

* https://ocio-jira.acf.hhs.gov/browse/TTAHUB-0=1857

### Every PR

- [x] Meets issue criteria
- [x] JIRA ticket status updated
- [x] Code is meaningfully tested
- [x] Meets accessibility standards (WCAG 2.1 Levels A, AA)
- [ ] API Documentation updated
- [ ] Boundary diagram updated
- [ ] Logical Data Model updated
- [ ] [Architectural Decision Records](https://adr.github.io/) written for major infrastructure decisions
- [ ] UI review complete

### Production Deploy

- [ ] Staging smoke test completed

### After merge/deploy

- [x] Update JIRA ticket status

### CODE REVIEW

File: frontend/src/pages/TrainingReports/components/EventCard.js [hunk 1]
```diff
@@ -19,12 +19,20 @@ function EventCard({
...
-  const isCollaborator = event.pocId && event.pocId.includes(user.id);
-  const canEditExisting = hasEditPermissions || (isCollaborator);
+
+  // Check if user has been assigned an event role.
+  const hasEditRole = (event.pocId && event.pocId.includes(user.id))
+    || (event.collaboratorIds && event.collaboratorIds.includes(user.id))
+    || (event.ownerId === user.id);
+
+  // Has region permissions and has been assigned an event role.
+  const canEditEvent = hasEditPermissions && hasEditRole;
```
Comment (thewatermethod) [inline frontend/src/pages/TrainingReports/components/EventCard.js:35]: We need to amend this. The poc (someone who only has collaborator permissions) needs to be able to create a session.

Comment (thewatermethod) [reply frontend/src/pages/TrainingReports/components/EventCard.js:35]: We'll probably want a FE test verifying that a collaborator sees the "create session" button as well

Comment (thewatermethod) [reply frontend/src/pages/TrainingReports/components/EventCard.js:35]: Same user also can't create a session if they visit the url directly, so the backend permissions are also wrong

Comment (thewatermethod) [reply frontend/src/pages/TrainingReports/components/EventCard.js:35]: The original chart that we're basing all this on is here:

https://app.mural.co/t/adhoccorporateworkspace2583/m/adhoccorporateworkspace2583/1683896571063/bc580f5a6ffa94c447fd8ad38f5535b5c8813ab6?invited=true&sender=u775286a3a84f5c5204174211

**Comment by thewatermethod** [comment]
I see the menu item now as a poc collaborator! that's good

Permissions for poc collaborators still appear to be wrong. To recreate: add a user as a poc collaborator to a report, impersonate that user and try to add a session. The backend returns 403.

File: src/policies/sessionReport.js [hunk 2]
```diff
@@ -55,12 +55,8 @@ export default class SessionReport {
   }
 
   canUpdate() {
-    if (!this.canWriteInRegion()) { return false; }
-
     if (this.isAdmin()) { return true; }
-    // if (this.isCollaborator()) { return true; }
     if (this.isAuthor()) { return true; }
```
Comment (thewatermethod) [inline src/policies/sessionReport.js:59]: POCs can edit all sessions on an event.

Comment (thewatermethod) [reply src/policies/sessionReport.js:59]: Please think about it this way:

- An owner can add collabs & POC to an event
- At that point, all 3 can create sessions, edit event, and edit all sessions.

Comment (AdamAdHocTeam) [reply src/policies/sessionReport.js:59]: Ok this has been updated.

Comment (thewatermethod) [reply src/policies/sessionReport.js:59]: A POC should be able to edit an event:

Screenshot 2023-07-13 at 4 48 30 PM

Comment (thewatermethod) [reply src/policies/sessionReport.js:59]: Other issues I see trying to edit as a POC:
- /training-report-users/ now throws a 403 so when editing an event as POC you'll see ```Error fetching collaborators and points of contact```
- /session-reports/eventId/1 also throws a 403 so when editing an event as POC you'll see ```Unable to load sessions```

What we want is:
- An owner can add collabs & POC to an event
- At that point, all 3 (owner, POC, and collabs) can create sessions, edit event, and edit all sessions.

**Review by thewatermethod — APPROVED** [review]
looking good

**Pull request merged**